src/main/kotlin/io/snyk/plugin/SnykProjectManagerListener.kt

```diff
@@ -7,18 +7,21 @@ import com.intellij.openapi.progress.Task.Backgroundable
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.project.ProjectManagerListener
 import snyk.common.lsp.LanguageServerWrapper
+import java.util.concurrent.ExecutorService
 import java.util.concurrent.Executors
 import java.util.concurrent.TimeUnit
 
 private const val TIMEOUT = 1L
 
 class SnykProjectManagerListener : ProjectManagerListener {
+    val threadPool: ExecutorService = Executors.newWorkStealingPool()
+
     override fun projectClosing(project: Project) {
         val closingTask = object : Backgroundable(project, "Project closing ${project.name}") {
             override fun run(indicator: ProgressIndicator) {
                 // limit clean up to 5s
                 try {
-                    Executors.newCachedThreadPool().submit {
+                    threadPool.submit {
```

Comment:
when we shut down, we don't want to create a new thread pool, we just want to use one.